### PR TITLE
Detect bidir 5665 v11

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5665

Describe changes:
- allows bidirectional signature matching !

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1889

Draft again because of second commit
General feedback expected :-)

TODO : 
- Give me better names
- Where should I document these new keywords `bidir.toclient` ?
- more tests !!!! Throw me rules examples ! negative and positive...

@victorjulien is there now a protocol with bidirectional frames ? (Enip rust has it in https://github.com/OISF/suricata/pull/11184)

https://github.com/OISF/suricata/pull/10904 rebased with some new test in SV

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20971

File: src/detect-engine-state.c [hunk 1]
```diff
@@ -239,17 +243,13 @@ void DetectRunStoreStateTx(
...
-        s->dir_state[1].filestore_cnt = 0;
-        s->dir_state[1].flags = 0;
-        /* reset 'cur' back to the list head */
-        s->dir_state[1].cur = s->dir_state[1].head;
+        for (int i = 0; i < DETECT_ENGINE_STATE_DIRECTIONS; i++) {
+            s->dir_state[i].cnt = 0;
+            s->dir_state[i].filestore_cnt = 0;
+            s->dir_state[i].flags = 0;
+            /* reset 'cur' back to the list head */
+            s->dir_state[i].cur = s->dir_state[0].head;
```
Comment (victorjulien) [inline src/detect-engine-state.c:251]: `dir_state[i]`?

**Comment by catenacyber** [comment]
Try to look at `git grep flowbits tests/test-ruleparse-etopen-01/emerging-all.rules | grep -v ":#" | grep -v "alert http " | grep -v "alert tcp " | grep -v "alert tcp-pkt" | grep -v "alert udp "`

File: src/detect-engine-state.c (hunk 1 above)
Comment (catenacyber) [reply src/detect-engine-state.c]: Thanks for the detail, nice catch.

Hope to read something more about the overview of this PR ;-)

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 21010

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/11293
